monai/fl/client/monai_algo.py

```diff
@@ -88,8 +91,242 @@ def disable_ckpt_loaders(parser):
+
+        Returns:
+            stats: ExchangeObject with summary statistics.
+
+        """
+
+        if self.dataset_root:
+            self.phase = FlPhase.GET_DATA_STATS
+            self.logger.info(f"Computing statistics on {self.dataset_root}")
+
+            if FlStatistics.HIST_BINS not in extra:
+                raise ValueError("FlStatistics.NUM_OF_BINS not specified in `extra`")
+            else:
+                hist_bins = extra[FlStatistics.HIST_BINS]
+            if FlStatistics.HIST_RANGE not in extra:
+                raise ValueError("FlStatistics.HIST_RANGE not specified in `extra`")
+            else:
+                hist_range = extra[FlStatistics.HIST_RANGE]
+
+            stats_dict = {}
+
+            # train data stats
+            train_summary_stats, train_case_stats = self._get_data_key_stats(
+                parser=self.train_parser,
+                data_key=self.train_data_key,
+                hist_bins=hist_bins,
+                hist_range=hist_range,
+                output_path=os.path.join(self.app_root, "train_data_stats.yaml"),
+            )
+            if train_case_stats:
+                # Only return summary statistics to FL server
+                stats_dict.update({self.train_data_key: train_summary_stats})
+
+            # eval data stats
+            eval_summary_stats, eval_case_stats = self._get_data_key_stats(
+                parser=self.train_parser,
+                data_key=self.eval_data_key,
+                hist_bins=hist_bins,
+                hist_range=hist_range,
+                output_path=os.path.join(self.app_root, "eval_data_stats.yaml"),
+            )
+            if eval_summary_stats:
+                # Only return summary statistics to FL server
+                stats_dict.update({self.eval_data_key: eval_summary_stats})
+
+            # total stats
+            if train_case_stats and eval_case_stats:
+                # Compute total summary
+                total_summary_stats = self._compute_total_stats(
+                    [train_case_stats, eval_case_stats], hist_bins, hist_range
+                )
+                stats_dict.update({FlStatistics.TOTAL_DATA: total_summary_stats})
+
+            # optional filter of data stats
+            stats = ExchangeObject(statistics=stats_dict)
+            if self.post_statistics_filters is not None:
+                for _filter in self.post_statistics_filters:
+                    stats = _filter(stats, extra)
+
+            return stats
+        else:
+            raise ValueError("data_root not set!")
+
+    def _get_data_key_stats(self, parser, data_key, hist_bins, hist_range, output_path=None):
+        if data_key not in parser:
+            self.logger.warning(f"Data key {data_key} not available in bundle configs.")
+            return None, None
+        data = parser.get_parsed_content(data_key)
+
+        datalist = {data_key: data}
+
+        analyzer = DataAnalyzer(
```

Comment:
Is it certain that only the image histogram analysis will be used for `MonaiAlgoStats`? If that's the case, the other operations in the `DataAnalyzer` or `SegSummarizer` do not contribute to `MonaiAlgoStats`, but it takes lots of computation that would slow things down. 

If it is needed, I would be happy to submit commits to your branch to illustrate how to utilize the Histogram Analyzers without wasting the time on other unnecessary operations.

Reply:
I would like to keep the other stats as well.

Reply:
Hi @mingxin-zheng, I'm merging this PR as it's good in general, but your point is also valid, please feel free to submit a follow up PR to revise this if needed...

Reply:
Thanks @holgerroth @wyli . I will submit a PR to make some analyzers optional.